`MyClass.schema()` returns `SchemaF[Any]`, providing no type safety, rather than `SchemaF[MyClass]`

This should give a mypy error, but doesn’t:

```python3
from dataclasses import dataclass
from dataclasses_json import DataClassJsonMixin

@dataclass
class MyClass(DataClassJsonMixin):
    my_field: int

x: str = MyClass.schema().load({"my_field": 0})
print(x.lower())
```

In reality, of course, `x` is a `MyClass`, but mypy doesn’t know that.

That’s because of this incorrect usage of a generic type alias:

https://github.com/lidatong/dataclasses-json/blob/3264a0046e1aa3c0a813335286ebdbc651f58b13/dataclasses_json/mm.py#L223 https://github.com/lidatong/dataclasses-json/blob/3264a0046e1aa3c0a813335286ebdbc651f58b13/dataclasses_json/api.py#L78-L88

A type alias `SchemaType = SchemaF[A]` needs to be used as `SchemaType[A]`, not `SchemaType`. The latter is interpreted as `SchemaType[Any]`. This is flagged by mypy with the [`disallow_any_generics` option](https://mypy.readthedocs.io/en/stable/config_file.html#confval-disallow_any_generics)—there are many other mypy errors, but here are the ones about `SchemaType`:

```
dataclasses_json/mm.py:322: error: Missing type parameters for generic type "SchemaType"
dataclasses_json/mm.py:362: error: Missing type parameters for generic type "SchemaType"
dataclasses_json/api.py:88: error: Missing type parameters for generic type "SchemaType"
```

It looks like this was broken by 9092e6f91235ed2468a5e0476f4117586c117fba (#96) in the name of Python 3.6 compatibility. If there is a compatibility problem with old Python versions, a better workaround is to quote the annotation to prevent runtime evaluation: `-> "SchemaType[A]"`.

--- a/dataclasses_json/api.py
+++ b/dataclasses_json/api.py
@@ -85,7 +85,7 @@ class DataClassJsonMixin(abc.ABC):
                load_only=(),
                dump_only=(),
                partial: bool = False,
-               unknown=None) -> SchemaType:
+               unknown=None) -> "SchemaType[A]":
         Schema = build_schema(cls, DataClassJsonMixin, infer_missing, partial)
 
         if unknown is None:

--- a/dataclasses_json/mm.py
+++ b/dataclasses_json/mm.py
@@ -140,7 +140,7 @@ TEncoded = typing.Dict[str, typing.Any]
 TOneOrMulti = typing.Union[typing.List[A], A]
 TOneOrMultiEncoded = typing.Union[typing.List[TEncoded], TEncoded]
 
-if sys.version_info >= (3, 7):
+if sys.version_info >= (3, 7) or typing.TYPE_CHECKING:
     class SchemaF(Schema, typing.Generic[A]):
         """Lift Schema into a type constructor"""
 
@@ -319,7 +319,7 @@ def schema(cls, mixin, infer_missing):
 def build_schema(cls: typing.Type[A],
                  mixin,
                  infer_missing,
-                 partial) -> typing.Type[SchemaType]:
+                 partial) -> typing.Type["SchemaType[A]"]:
     Meta = type('Meta',
                 (),
                 {'fields': tuple(field.name for field in dc_fields(cls)
@@ -359,7 +359,7 @@ def build_schema(cls: typing.Type[A],
         return dumped
 
     schema_ = schema(cls, mixin, infer_missing)
-    DataClassSchema: typing.Type[SchemaType] = type(
+    DataClassSchema: typing.Type["SchemaType[A]"] = type(
         f'{cls.__name__.capitalize()}Schema',
         (Schema,),
         {'Meta': Meta,

